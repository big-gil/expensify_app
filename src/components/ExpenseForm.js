import React from "react";
import moment from "moment";
// import { SingleDatePicker } from "react-dates";
import { connect } from "react-redux";

const now = moment().valueOf();

export default class ExpenseForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      description: props.expense ? props.expense.description : "",
      note: props.expense ? props.expense.note : "",
      amount: props.expense ? (props.expense.amount / 100).toString() : "",
      createDate: props.expense ? props.expense.createDate : now,
      error: ""
    };
  }

  onDescriptionChange = e => {
    const description = e.target.value;
    this.setState(() => ({ description }));
  };

  onNoteChange = e => {
    const note = e.target.value;
    this.setState(() => ({ note }));
  };

  onAmountChange = e => {
    const amount = e.target.value;
    if (!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)) {
      this.setState(() => ({ amount }));
    }
  };

  onDateChange = e => {
    const createDate = e.target.value;
    this.setState(() => ({ createDate }));
  };

  onSubmitForm = e => {
    e.preventDefault();

    if (!this.state.description || !this.state.amount) {
      this.setState(() => ({
        error: "Please enter a description and an amount."
      }));
    } else {
      this.setState(() => ({ error: "" }));
      this.props.onSubmit({
        description: this.state.description,
        amount: Math.round(parseFloat(this.state.amount, 10) * 100),
        note: this.state.note,
        createDate: this.state.createDate
      });
    }
  };

  render() {
    return (
      <div>
        {this.state.error && <p>{this.state.error}</p>}
        <form onSubmit={this.onSubmitForm}>
          <input
            type="text"
            placeholder="Description"
            autoFocus
            value={this.state.description}
            onChange={this.onDescriptionChange}
          />
          <input
            type="text"
            placeholder="Amount"
            value={this.state.amount}
            onChange={this.onAmountChange}
          />
          <input
            type="number"
            placeholder="Date"
            value={this.state.createDate}
            onChange={this.onDateChange}
          />
          <textarea
            value={this.state.note}
            placeholder="Add a note for your expense (optional)"
            onChange={this.onNoteChange}
          />
          <input type="submit" value="Add Expense" />
        </form>
      </div>
    );
  }
}
