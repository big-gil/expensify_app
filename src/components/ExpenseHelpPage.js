import React from "react";

const ExpenseHelpPage = () => (
  <div>
    <p>This is from Expense Help Page</p>
  </div>
);

export default ExpenseHelpPage;
