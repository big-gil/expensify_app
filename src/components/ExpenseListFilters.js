import React from "react";
import { connect } from "react-redux";
import { setTextFilter, sortByAmount, sortByDate } from "../actions/filter";

const ExpenseListFilters = props => (
  <div>
    <input
      type="text"
      value={props.filters.text}
      onChange={e => {
        props.dispatch(setTextFilter(e.target.value));
      }}
    />
    <select
      value={props.filters.text}
      onChange={e => {
        {
          e.target.value === "amount" && props.dispatch(sortByAmount());
        }
        {
          e.target.value === "date" && props.dispatch(sortByDate());
        }
      }}
    >
      <option value="date">date</option>
      <option value="amount">amount</option>
    </select>
  </div>
);

const mapStateToProps = state => {
  return {
    filters: state.filters
  };
};

export default connect(mapStateToProps)(ExpenseListFilters);
