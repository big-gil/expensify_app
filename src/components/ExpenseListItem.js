import React from "react";
import { Link } from "react-router-dom";
import moment from "moment";

const ExpenseListItem = ({ id, description, amount, createDate }) => (
  <div>
    <Link to={`/edit/${id}`}>
      <h3>{description}</h3>
    </Link>

    <p>
      {amount} - {moment(createDate).format("MM/DD/YYYY")}
    </p>
  </div>
);

export default ExpenseListItem;
