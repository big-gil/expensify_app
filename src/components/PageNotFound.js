import React from "react";
import { Link } from "react-router-dom";

const PageNotFound = () => (
  <div>
    <p>
      404! Page not found. <Link to="/">Go To Home Page</Link>
    </p>
  </div>
);

export default PageNotFound;
