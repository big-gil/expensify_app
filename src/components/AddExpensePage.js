import React from "react";
import ExpenseForm from "./ExpenseForm";
import { connect } from "react-redux";
import { addExpense } from "../actions/expense";
import { withRouter } from "react-router-dom";

const AddExpensePage = props => (
  <div>
    <h2>Add Expense</h2>
    <ExpenseForm
      onSubmit={expense => {
        props.dispatch(addExpense(expense));
        props.history.push("/");
      }}
    />
  </div>
);

export default connect()(withRouter(AddExpensePage));
