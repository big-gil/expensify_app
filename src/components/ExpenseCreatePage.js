import React from "react";
import AddExpensePage from "./AddExpensePage";

const ExpenseCreatePage = () => (
  <div>
    <AddExpensePage />
  </div>
);

export default ExpenseCreatePage;
