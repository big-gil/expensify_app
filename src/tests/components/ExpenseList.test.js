import { ExpenseList } from "../../components/ExpenseList";
import { shallow } from "enzyme";
import React from "react";
import { expenses } from "../fixtures/expenses";

test("Expense List Test 1 - Expense List", () => {
  const wrapper = shallow(<ExpenseList expenses={expenses} />);
  expect(wrapper).toMatchSnapshot();
});

test("Expense List Test 2 - No Expenses To List", () => {
  const wrapper = shallow(<ExpenseList expenses={[]} />);
  expect(wrapper).toMatchSnapshot();
});
