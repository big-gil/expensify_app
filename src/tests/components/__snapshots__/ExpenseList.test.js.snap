// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expense List Test 1 - Expense List 1`] = `
<div>
  <h1>
    Expense List
  </h1>
  <ExpenseListItem
    amount={14500}
    createDate={604800000}
    descripton="Water Bill"
    id={
      Any {
        "$$typeof": Symbol(jest.asymmetricMatcher),
        "sample": [Function],
      }
    }
    key="Any"
    note="note1"
  />
  <ExpenseListItem
    amount={500000}
    createDate={-259200000}
    descripton="Gas Bill"
    id={
      Any {
        "$$typeof": Symbol(jest.asymmetricMatcher),
        "sample": [Function],
      }
    }
    key="Any"
    note="note2"
  />
  <ExpenseListItem
    amount={122000}
    createDate={172800000}
    descripton="Rent"
    id="123abc"
    key="123abc"
    note="note3"
  />
</div>
`;

exports[`Expense List Test 2 - No Expenses To List 1`] = `
<div>
  <h1>
    Expense List
  </h1>
  <p>
    No Expenses
  </p>
</div>
`;
